arcaflow_plugin_fio/fio_plugin.py

```diff
@@ -17,6 +17,23 @@
 )
 
 
+def split_json_and_errors(fio_output: str):
+    # Sometimes fio informational messages are included at the top of the JSON output.
+    # Separate the json data from the errors and return them as a tuple.
+    error_data = ""
+    lines = fio_output.splitlines()
+    for i in range(len(fio_output)):
+        file_data = "\n".join(lines[i:])
+        try:
+            json_data = json.loads(file_data)
+            break
+        except json.JSONDecodeError:
+            error_data += f"{lines[i]}\n"
+            continue
+
+    return json_data, error_data
```

Comment:
This should fail, as `json_data` is undefined, if the loop doesn't successfully `loads` any JSON. Is this a realistic possibility? Seems it'd make sense to allow for that, at least...

Reply:
When there is a failure, oddly enough it always still generates the json structure, just with empty/0 values.

Reply:
"When there is a failure, oddly enough (the current version) always (seems to in our experience) generate[s]" ...

I don't see any reason for the wrapper code here to be dependent on an assumption of that behavior, and I'd prefer that it wasn't? 😆